### PR TITLE
New harmonic

### PR DESCRIPTION
**Pull request by lmalina**
Added new mode in harpy 'window':
It uses nuttal4 window on SVD decomposed data and zero-pads to 2^21
then it recomposes the coefficients in the frequency domain

### CODE REVIEW

File: harmonic_analysis/harpy.py [hunk 1]
```diff
@@ -521,4 +515,48 @@ def _which_compute_coef():
...
 _compute_coef = _which_compute_coef()
+
+
+def windowed_padded_fft(matrix, svd, turn_bits, harpy_input):
+    # TODO fft is used just once on real data -> rfft can be used and together with np.conj() when needed
+    for_freqs = np.dot(np.diag(svd[1]), svd[2])
+    length = for_freqs.shape[1]
+    padded_length = np.power(2, turn_bits)
+    ints2pi = 2. * np.pi * np.arange(length) / float(length)
+    nuttal4 = 0.3125 - 0.46875 * np.cos(ints2pi) + 0.1875 * np.cos(2. * ints2pi) - 0.03125 * np.cos(3. * ints2pi)
```
Comment (jaimecp89) [inline harmonic_analysis/harpy.py]: This should be a constant, maybe a constant dictionary with different windows?

File: harmonic_analysis/harpy.py [hunk 2]
```diff
@@ -521,4 +515,48 @@ def _which_compute_coef():
...
+    norm = np.sum(nuttal4)
+    #nuttal3 = 0.375 - 0.5 * np.cos(ints2pi) + 0.125 * np.cos(2. * ints2pi)
+    #norm = np.sum(nuttal3)
```
Comment (jaimecp89) [inline harmonic_analysis/harpy.py]: Cmon, dont let comments!

File: harmonic_analysis/harpy.py [hunk 3]
```diff
@@ -66,12 +60,12 @@ def harpy(harpy_input, bpm_matrix_x, usv_x, bpm_matrix_y, usv_y):
...
     for plane, bpm_matrix, usv in cases:
         panda = pd.DataFrame(index=bpm_matrix.index, columns=OrderedDict())
-        frequencies, coefficients = harmonic_analysis(
-            bpm_matrix,
-            usv=usv,
-            mode=harpy_input.harpy_mode,
-            sequential=harpy_input.sequential,
-        )
+        if harpy_input.harpy_mode == "window":
+            frequencies, coefficients = windowed_padded_fft(bpm_matrix, usv, 21, harpy_input)
```
Comment (jaimecp89) [inline harmonic_analysis/harpy.py]: This 21 is a magic number (-ish) probably it should go in a constant.

File: harmonic_analysis/harpy.py (hunk 1 above)
Comment (lmalina) [reply harmonic_analysis/harpy.py]: Other windows should be studied ... later on

**Pull request merged**